omc3/optics_measurements/beta_from_amplitude.py

```diff
@@ -65,8 +65,7 @@ def beta_from_amplitude(meas_input, input_files, plane, tunes):
     df = pd.DataFrame(meas_input.accelerator.model).loc[:, ["S", f"MU{plane}", f"BET{plane}"]]
     df.rename(columns={f"MU{plane}": f"MU{plane}{MDL}",
                        f"BET{plane}": f"BET{plane}{MDL}"}, inplace=True)
-    dpp_value = meas_input.dpp if "dpp" in meas_input.keys() else 0
-    df = pd.merge(df, input_files.joined_frame(plane, [f"AMP{plane}", f"MU{plane}"], dpp_value=dpp_value),
+    df = pd.merge(df, input_files.joined_frame(plane, [f"AMP{plane}", f"MU{plane}"], dpp_value=meas_input.analyse_dpp),
```

Comment:
yes and no. This was a very old and non-straighforward way to change the dpp used in the analysis, which made sense for the chromatic beating but when I rewrote it to use `analyse_dpp` I missed this part (because I didn't know about this). So the other functions were not using this parameter anymore anyway, and maybe this is also why the W was completely 0: it was always using dpp of 0 instead of the provided one in the chromatic beating function.

Reply:
Ok, good catch.